<!-- qr test launch file -->
<launch>
    <node pkg="mr_camera_processing" name="CameraProcessing" type="mr_camera_processing" output="screen">
	<param name="sub_image" value="/mrCamera/image" />
	<param name="pub_cross" value="/mrCameraProcessing/cross" />
	<param name="pub_qr" value="/mrCameraProcessing/QR" />
	<param name="pub_line" value="/mrCameraProcessing/line" />
	<param name="pub_image" value="/mrCameraProcessing/output_image" />
	<param name="srv_enable" value="/mrCameraProcessing/enable" />
  	<param name="QR_min_white_area" value= "2000.0" />
	<param name="QR_min_black_area" value= "11" />
	<param name="QR_grayscale_threshold" value= "162.0" />
    </node>

   <node pkg="mr_line_follower" name="LineFollower" type="mr_line_follower" output="screen">
	<param name="pid_p" value="1.3" />
	<param name="pid_i" value="0.0015" />
	<param name="pid_d" value="3.8" />
	<param name="pid_dt" value="0.33" />
	<param name="pid_max" value="600" />
	<param name="pid_min" value="-600" />
	<param name="reference_point_x" value="320" />
	<param name="reference_point_y" value="240" />
	<param name="robot_speed_lidar" value="0.2" />
	<param name="robot_speed_qr" value="0.43" />
	<param name="robot_speed_qr_slow" value="0.1" />
	<param name="robot_speed_rel" value="0.2" />
	<param name="robot_turn_speed" value="1.2" />
	<param name="robot_turn_speed_qr_slow" value="0.3" />
	<param name="lidar_distance" value="0.2" />
	<param name="relative_distance" value="0.2" />
	<param name="linear_precision" value="0.005" />
	<param name="srv_mr_camera_processing_enable_name" value="/mrCameraProcessing/enable" />
	<param name="sub_cross" value="/mrCameraProcessing/cross" />
	<param name="sub_line" value="/mrCameraProcessing/line" />
	<param name="sub_lidar" value="/scan" />
	<param name="sub_odom" value="/fmKnowledge/pose" />
	<param name="sub_qr" value="/mrCameraProcessing/QR" />
	<param name="pub_twist" value="/fmCommand/cmd_vel" />
	<param name="pub_deadman" value="/fmSafe/deadman" />
    </node>
	
</launch>

